netcore/System.Private.CoreLib/src/System/Threading/LowLevelLifoSemaphore.Unix.Mono.cs

```diff
@@ -31,14 +31,18 @@ public void Dispose ()
 		{
 		}
 
+		// Do not inline, to ensure icall handles are references to locals and native code can omit barriers.
+		//
+		[MethodImplAttribute (MethodImplOptions.NoInlining)]
 		private bool WaitCore (int timeoutMs)
 		{
 			WaitEntry wait_entry = new WaitEntry ();
 			bool mutexLocked = false;
 			bool waitEntryLocked = false;
+			bool allowInterruption = false;
 
 			try {
-				Monitor.try_enter_with_atomic_var (mutex, Timeout.Infinite, false, ref mutexLocked);
+				Monitor.try_enter_with_atomic_var (ref mutex, Timeout.Infinite, allowInterruption, ref mutexLocked);
```

Comment:
`ref mutex` here is a ref to a field in `this` (which is on the heap), so we don't actually have a guarantee that whatever `mutex` points to is pinned.

Reply:
I didn't realize this, but, really? It doesn't suffice?
Solution is to copy it to a local and ref that I guess, and then no need to answer prior question.